gui/src/shortcutdialog.cpp

```diff
@@ -0,0 +1,311 @@
+    static const QList<QPair<ChiakiScreenModePreset, const char *>> mode_strings = {
+        { CHIAKI_MODE_STRETCH, "stretch" },
+        { CHIAKI_MODE_ZOOM, "zoom" },
+        { CHIAKI_MODE_FULLSCREEN, "fullscreen" }
+    };
+    for(const auto &p : mode_strings)
+    {
+        Ui::ShortcutDialog::mode_combo_box->addItem(tr(p.second), p.first);
+    }
+
+    //Widgets
+    artworkWidgets.insert(ArtworkType::LANDSCAPE,
+        new SGDBArtworkWidget(this, &log, ArtworkType::LANDSCAPE, landscape_label, landscape_upload_button, landscape_prev_button, landscape_next_button, landscape_game_combo));
+    artworkWidgets.insert(ArtworkType::PORTRAIT,
+        new SGDBArtworkWidget(this, &log, ArtworkType::PORTRAIT, portrait_label, portrait_upload_button, portrait_prev_button, portrait_next_button, portrait_game_combo));
+    artworkWidgets.insert(ArtworkType::HERO,
+        new SGDBArtworkWidget(this, &log, ArtworkType::HERO, hero_label, hero_upload_button, hero_prev_button, hero_next_button, hero_game_combo));
+    artworkWidgets.insert(ArtworkType::ICON,
+        new SGDBArtworkWidget(this, &log, ArtworkType::ICON, icon_label, icon_upload_button, icon_prev_button, icon_next_button, icon_game_combo));
+    artworkWidgets.insert(ArtworkType::LOGO,
+        new SGDBArtworkWidget(this, &log, ArtworkType::LOGO, logo_label, logo_upload_button, logo_prev_button, logo_next_button, logo_game_combo));
+
+    //Shortcut Button
+    connect(Ui::ShortcutDialog::add_to_steam_button, &QPushButton::clicked, [=]() {
+        std::map<std::string, std::string> artwork;
+        artwork["landscape"] = artworkWidgets.value(ArtworkType::LANDSCAPE)->getUrl().toStdString();
+        artwork["portrait"] = artworkWidgets.value(ArtworkType::PORTRAIT)->getUrl().toStdString();
+        artwork["hero"] = artworkWidgets.value(ArtworkType::HERO)->getUrl().toStdString();
+        artwork["logo"] = artworkWidgets.value(ArtworkType::LOGO)->getUrl().toStdString();
+        artwork["icon"] = artworkWidgets.value(ArtworkType::ICON)->getUrl().toStdString();
+        CreateShortcut(server, artwork);
+    });
+
+    //QTimer::singleShot(0, this, &ShortcutDialog::dialogLoaded);
+}
+
+void ShortcutDialog::ExternalChanged() {
+    Ui::ShortcutDialog::external_dns_edit->setFocus();
+    Ui::ShortcutDialog::external_dns_edit->setEnabled(Ui::ShortcutDialog::allow_external_access_checkbox->isChecked());
+    Ui::ShortcutDialog::local_ssid_edit->setEnabled(Ui::ShortcutDialog::allow_external_access_checkbox->isChecked());
+}
+
+std::string ShortcutDialog::getExecutable() {
+#if defined(__APPLE__)
+    char buffer[PATH_MAX];
+    uint32_t size = sizeof(buffer);
+
+    if (_NSGetExecutablePath(buffer, &size) == -1) {
+        // Buffer is not large enough, allocate a larger one
+        char* largerBuffer = new char[size];
+        if (_NSGetExecutablePath(largerBuffer, &size) == 0) {
+            return largerBuffer;
+        }
+    } else {
+        return buffer;
+    }
+#elif defined(__linux)
```

Comment:
This part is all wrong. First it assumes the flatpak name and the other path won't work for appimage either I think (it would return the executable path inside the temporary appimage mount).